Sources/StreamChat/Controllers/ChannelController/LivestreamChannelController.swift

```diff
@@ -0,0 +1,502 @@
+            handleNewReaction(reactionNewEvent)
+            
+        case let reactionUpdatedEvent as ReactionUpdatedEvent:
+            handleUpdatedReaction(reactionUpdatedEvent)
+            
+        case let reactionDeletedEvent as ReactionDeletedEvent:
+            handleDeletedReaction(reactionDeletedEvent)
+            
+        default:
+            break
+        }
+    }
+    
+    private func handleNewMessage(_ message: ChatMessage) {
+        var currentMessages = messages
+
+        // If message already exists, update it instead
+        if currentMessages.contains(where: { $0.id == message.id }) {
+            handleUpdatedMessage(message)
+            return
+        }
+
+        currentMessages.insert(message, at: 0)
+        messages = currentMessages
+
+        notifyDelegateOfChanges()
+    }
+    
+    private func handleUpdatedMessage(_ updatedMessage: ChatMessage) {
+        var currentMessages = messages
+
+        if let index = currentMessages.firstIndex(where: { $0.id == updatedMessage.id }) {
+            currentMessages[index] = updatedMessage
+            messages = currentMessages
+
+            notifyDelegateOfChanges()
+        }
+    }
+    
+    private func handleDeletedMessage(_ deletedMessage: ChatMessage) {
+        var currentMessages = messages
+
+        currentMessages.removeAll { $0.id == deletedMessage.id }
+        messages = currentMessages
+
+        notifyDelegateOfChanges()
+    }
+
+    private func handleNewReaction(_ reactionEvent: ReactionNewEvent) {
+        updateMessage(reactionEvent.message)
+    }
+    
+    private func handleUpdatedReaction(_ reactionEvent: ReactionUpdatedEvent) {
+        updateMessage(reactionEvent.message)
+    }
+    
+    private func handleDeletedReaction(_ reactionEvent: ReactionDeletedEvent) {
+        updateMessage(reactionEvent.message)
+    }
+    
+    private func updateMessage(
+        _ updatedMessage: ChatMessage
+    ) {
+        let messageId = updatedMessage.id
+        var currentMessages = messages
+
+        guard let messageIndex = currentMessages.firstIndex(where: { $0.id == messageId }) else {
+            return
+        }
+
+        currentMessages[messageIndex] = updatedMessage
+        messages = currentMessages
+
+        notifyDelegateOfChanges()
+    }
+    
+    private func notifyDelegateOfChanges() {
+        guard let channel = channel else { return }
+
+        delegate?.livestreamChannelController(self, didUpdateChannel: channel)
+        delegate?.livestreamChannelController(self, didUpdateMessages: messages)
+    }
+}
+
+// MARK: - Delegate Protocol
+
+/// Delegate protocol for `LivestreamChannelController`
+public protocol LivestreamChannelControllerDelegate: AnyObject {
+    /// Called when the channel data is updated
+    /// - Parameters:
+    ///   - controller: The controller that updated
+    ///   - channel: The updated channel the controller manages.
+    func livestreamChannelController(
```

Comment:
maybe we keep the same delegate method signatures as the regular controller, to make the integration easier?